### PR TITLE
Suppressed pylint warning

### PR DESCRIPTION
**Pull request by galbramc**
Pylint was upgraded to version 1.7.2, and introduced new warnings.

### CODE REVIEW

**Comment by galbramc** [comment]
@bqpd and @whoburg, we had a hard dive failure on one of our testing machines. This forced us to replace the box and reinstall everything. In the process we got a newer version of pylint, which came with some new warnings. Trying to fix them here as best as I can, but I can't fix them all.

**Comment by galbramc** [comment]
I did not actually mind code that looked like this:

```
if this: return this
else: return that
```
but if you prefer that pylint tells you not to do that.

**Comment by whoburg** [comment]
That only occurred in one place. The bigger style issue is what to do with code like

`if this:
    return that
else:
    return other`

Which the new pylint complains about. In this PR I've changed all those to either

`return this if that else other`,

when the line is long enough, or

```python
if this:
    return that
return other
```
when the lines are long.
@bqpd do you feel strongly or shall we just follow pylint's rules now that I've made the changes?

**Comment by whoburg** [comment]
@bqpd, ready for review.

**Comment by whoburg** [comment]
@galbramc I can't figure out which tests are failing. Tests pass on my machine.

**Comment by whoburg** [comment]
@bqpd another issue is a lot of string comparisons using `is`, which this PR changes to `==`. Were those done for speed?

**Comment by galbramc** [comment]
Here are the errors:

/jenkins/workspace/gpkit_PullRequest_Unit/buildnode/reynolds/optimizer/mosek/gpkit/small_scripts.py:62: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison

/jenkins/workspace/gpkit_PullRequest_Unit/buildnode/reynolds/optimizer/mosek/gpkit/nomials/data.py:186: RuntimeWarning: invalid value encountered in double_scalars

**Comment by whoburg** [comment]
retest this please

**Comment by galbramc** [comment]
Are you trying to divide by zero? 

https://stackoverflow.com/questions/27784528/numpy-division-with-runtimewarning-invalid-value-encountered-in-double-scalars

**Comment by whoburg** [comment]
that `RuntimeWarning: invalid value encountered in double_scalars` is definitely an issue, but it's been around for awhile, and it doesn't cause the tests to fail. I still don't understand why they are being marked as failures because the console output shows them all passing.

**Comment by whoburg** [comment]
Jenkins shows "Test Result (no failures)" and then marks the test as failed. I don't understand. Worst user interface ever.

**Comment by galbramc** [comment]
It's the log parser that is marking the failure because it's parsing the console output. We were not looking for that RuntimWarning before, only the FutureWarning. I just added the RuntimWarning in the hopes you could fix it.

**Comment by whoburg** [comment]
```
Run condition [Execution node ] preventing perform for step [Execute Windows batch command]
Build step 'Console output (build log) parsing' changed build result to FAILURE
```
@galbramc I think we'll need your help understanding this.

**Comment by galbramc** [comment]
If you don't want to fix it we can continue to ignore the warning.

**Comment by galbramc** [comment]
Yeah that just means it's the log parser that found the error. The log parse errors are here:

https://acdl.mit.edu/csi/job/gpkit_PullRequest_Unit/2172/buildnode=reynolds,optimizer=cvxopt/parsed_console/

**Comment by whoburg** [comment]
The warning was being ignored all along. Did that suddenly change at the same time as the pylint upgrade? I'm trying to understand why we went from unit tests passing to unit tests failing.

**Comment by galbramc** [comment]
We were only looking for FutureWarnings, not RuntimeWarnings. I just added RuntimeWarning as one of the things to look for. I can remove it if you don't consider RuntimeWarnings an error.

**Comment by galbramc** [comment]
it actually happens in the beam.py example as well.

**Comment by whoburg** [comment]
I do consider it a problem, but we need some time to fix it and I don't think it's fair to just suddenly turn it on and have all our tests marked as failing. Let me make a new issue for that and we'll plan to turn on parsing for `RuntimeWarning`s as soon as we fix that issue.

**Comment by galbramc** [comment]
Ok I'll turn it off for now in the log parse, but I'll leave it in the email. So you will see it in the emails if some other error occurs.

**Comment by galbramc** [comment]
test gpkit please

**Comment by whoburg** [comment]
perfect thanks... see #1140

**Comment by galbramc** [comment]
test models please

**Comment by galbramc** [comment]
strange, turbofan only failed on one machine with mosek not finding an optimal solution:

https://acdl.mit.edu/csi/job/gpkit_PullRequest_Models/153/buildnode=windows7x64,optimizer=mosek/parsed_console/

**Comment by galbramc** [comment]
test models please

**Comment by galbramc** [comment]
same error, different machine... this is looking chaotic. Is mosek not deterministic?

https://acdl.mit.edu/csi/job/gpkit_PullRequest_Models/154/buildnode=macys,optimizer=mosek/parsed_console/

**Comment by whoburg** [comment]
Definitely seems to be an issue with turbofan but we've seen that in the past, https://github.com/hoburg/turbofan/issues/8. Not introduced by this PR.

**Comment by whoburg** [comment]
@bqpd ready for review and would like to merge soon to patch failing test badges on homepage.

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
test gpkit please

File: gpkit/_cvxopt.py [hunk 1]
```diff
@@ -253,7 +253,8 @@ def Fgp(x = None, z = None):
                 blas.syrk(Fsc, H, trans='T', k=stop-start, alpha=z[i],
                     beta=1.0)
 
-        if z is None: return f, Df
-        else: return f, Df, H
+        if z is None:
+            return f, Df
+        return f, Df, H
```
Comment (bqpd) [inline gpkit/_cvxopt.py]: Lol this is not our code :p

File: gpkit/build.py [hunk 2]
```diff
@@ -28,9 +28,8 @@ def isfile(path):
     if os.path.isfile(path):
         log("#     Found %s" % path)
         return True
-    else:
-        log("#     Could not find %s" % path)
-        return False
+    log("#     Could not find %s" % path)
+    return False
```
Comment (bqpd) [inline gpkit/build.py]: I slightly prefer the previous style, but don't much care.

File: gpkit/constraints/geometric_program.py [hunk 3]
```diff
@@ -388,7 +388,7 @@ def genA(exps, varlocs):
         for i in varlocs[var]:
             exp = exps[i][var]
             A.append(i, j, exp)
-            if varsign is "both":
+            if varsign == "both":
```
Comment (bqpd) [inline gpkit/constraints/geometric_program.py]: reasonable, using `is` for strings is a bit hacky

File: gpkit/constraints/prog_factories.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 try:
     from ipyparallel import Client
     CLIENT = Client(timeout=0.01)
-    assert len(CLIENT) > 0
+    assert len(CLIENT) > 0  # pylint:disable=len-as-condition
```
Comment (bqpd) [inline gpkit/constraints/prog_factories.py]: Huh that's an interesting new error.

File: gpkit/interactive/sensitivity_map.py [hunk 5]
```diff
@@ -134,15 +132,15 @@ def constraint_latex_list(self, paintby):
                 else:
                     constrs = [constr]
                 constr_texs = []
-                for constr in constrs:
```
Comment (bqpd) [inline gpkit/interactive/sensitivity_map.py]: @whoburg, what was the bug here?

File: gpkit/nomials/array.py [hunk 6]
```diff
@@ -61,23 +61,21 @@ def str_without(self, excluded=None):
...
-            return str(self.flatten()[0])  # TODO THIS IS WEIRD
+        return str(self.flatten()[0])  # TODO THIS IS WEIRD
 
     def latex(self, matwrap=True):
         "Returns 1D latex list of contents."
-        if len(self.shape) == 0:
+        if self.ndim == 0:
             return self.flatten()[0].latex()
-        if len(self.shape) == 1:
+        if self.ndim == 1:
```
Comment (bqpd) [inline gpkit/nomials/array.py]: Yeah, much better.

**Comment by bqpd** [comment]
Reviewed with one question. Ready to merge.

File: gpkit/constraints/prog_factories.py (hunk 4 above)
Comment (whoburg) [inline gpkit/constraints/prog_factories.py]: I think the more pythonic thing is for empty iterables to implement `__bool__` to return False, but I wasn't sure if `Client` implements that behavior.

File: gpkit/interactive/sensitivity_map.py (hunk 5 above)
Comment (whoburg) [inline gpkit/interactive/sensitivity_map.py]: overwriting the argument of the method

Comment (whoburg) [inline gpkit/interactive/sensitivity_map.py]: errr not the argument of the method but rather the iteration variable of the for loop

**Comment by whoburg** [comment]
test models please

**Pull request merged**